components/admin_console/admin_navbar_dropdown/index.js

```diff
@@ -13,10 +14,17 @@ import {getNavigationBlocked} from 'selectors/views/admin';
 import AdminNavbarDropdown from './admin_navbar_dropdown.jsx';
 
 function mapStateToProps(state) {
+    const teams = getMyTeams(state);
+    const redirectChannelPerTeam = {};
+    for (const team of Object.values(teams)) {
+        redirectChannelPerTeam[team.id] = getRedirectChannelNameForTeam(state, team.id);
+    }
+
     return {
         locale: getCurrentLocale(state),
-        teams: getMyTeams(state),
+        teams,
         navigationBlocked: getNavigationBlocked(state),
+        redirectChannelPerTeam,
```

Comment:
`redirectChannelPerTeam` will be a new object each time, so we should use a selector to make that if it's still needed